linera-views/src/reentrant_collection_view.rs

```diff
@@ -178,48 +178,83 @@ where
     ViewError: From<C::Error>,
     W: View<C> + Send + Sync,
 {
+    /// Reads the view and if missing returns the default view
     async fn wrapped_view(
         context: &C,
-        clear: bool,
+        delete_storage_first: bool,
         short_key: &[u8],
     ) -> Result<Arc<RwLock<W>>, ViewError> {
         let key = context.base_tag_index(KeyTag::Subview as u8, short_key);
         let context = context.clone_with_base_key(key);
         // Obtain a view and set its pending state to the default (e.g. empty) state
         let mut view = W::load(context).await?;
-        if clear {
+        if delete_storage_first {
             view.clear();
         }
         Ok(Arc::new(RwLock::new(view)))
     }
 
-    async fn try_load_view(
+    /// Reads the view if existing, otherwise report an error.
+    async fn wrapped_view_check(
         context: &C,
-        updates: &mut BTreeMap<Vec<u8>, Update<Arc<RwLock<W>>>>,
-        was_cleared: bool,
+        delete_storage_first: bool,
         short_key: &[u8],
-    ) -> Result<Arc<RwLock<W>>, ViewError> {
-        use btree_map::Entry::*;
+    ) -> Result<Option<Arc<RwLock<W>>>, ViewError> {
+        let key_index = context.base_tag_index(KeyTag::Index as u8, short_key);
+        if delete_storage_first || !context.contains_key(&key_index).await? {
+            Ok(None)
+        } else {
+            Ok(Some(
+                Self::wrapped_view(context, delete_storage_first, short_key).await?,
```

Comment:
Note that `delete_storage_first` is always false here.

Seems that we should not call this function in the first place when `delete_storage_first == true` and just assume `None`.

Reply:
That test for delete_storage_first is done just above so not sure what you mean.